UnchainedLauncherCore/src/Installer/UnchainedLauncherInstaller.cs

```diff
@@ -0,0 +1,267 @@
+
+                    log("Exitting current process to launch new launcher");
+                    EndProgram();
+                } else {
+                    MoveExistingLauncher(targetDir, log);
+
+                    log($"Replacing launcher \n    at {launcherPath} \n    with downloaded launcher from {downloadFilePath}");
+                    File.Move(downloadFilePath, launcherPath, true);
+                    log($"Successfully installed launcher version {release.Version}");
+                }
+
+
+                return true;
+            } catch (Exception ex) {
+                log(ex.ToString());
+                logger.Error(ex);
+            }
+
+            return false;
+        }
+
+        private static void MoveExistingLauncher(DirectoryInfo targetDir, Action<string> log) {
+            var launcherPath = Path.Combine(targetDir.FullName, FilePaths.LauncherPath);
+            var originalLauncherPath = Path.Combine(targetDir.FullName, FilePaths.OriginalLauncherPath);
+
+            log("Checking if the existing launcher needs to be moved...");
+
+            // Only if the Product Name of the file at the launcher path is not the same as the current executable
+            if (File.Exists(launcherPath)) {
+                var launcherProductName = FileVersionInfo.GetVersionInfo(launcherPath)?.ProductName;
+                var currentExecutableProductName = Assembly.GetExecutingAssembly().GetName().Name;
+
+                if (launcherProductName != currentExecutableProductName) {
+                    log($"Existing launcher is not {currentExecutableProductName}. Moving existing launcher to {originalLauncherPath}");
+                    File.Move(launcherPath, originalLauncherPath, true);
+                } else {
+                    log("Existing launcher is a modified launcher. Overwriting with version selected in the installer..");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Downloads the assets of a release to the specified download targets.
+        /// 
+        /// The assetDownloadTargets thing is probably overkill, but may come in handy if we ever split out the installer and launcher in to separate exes.
+        /// </summary>
+        /// <param name="download">The release to download files from</param>
+        /// <param name="assetDownloadTargets">A function returning Some(filePath) whenever an asset should be downloaded</param>
+        /// <returns></returns>
+        private static async Task<bool> DownloadRelease(VersionedRelease download, Func<ReleaseAsset, Option<string>> assetDownloadTargets, Action<string> log) {
+            var results =
+                await download.Release.Assets.ToList().Select(async asset => {
+                    var downloadTarget = assetDownloadTargets(asset);
+
+                    if (downloadTarget.IsNone) {
+                        logger.Debug($"Skipping asset with no download target {asset.Name}");
+                        return true;
```

Comment:
return false? otherwise assets with no target are marker?

Reply:
true/false here marks failure. There's no failure happening here. Should probably add that to the doc block